department-of-veterans-affairs/abd-vro
# MCP-2230 persist claim

## What was the problem?

1. There is a new requirement to persist v2 requests in the claim tables.
2. There is a bug: When the same claim is called with a different diagnostic code, a new contention is not created.

Associated tickets or Slack threads:

- [MCP-2230](https://amida.atlassian.net/browse/MCP-2230)

## How does this fix it?[^1]

1. Persist v2 requests in the claim tables.
2. Persist a different contention for each diagnostic code

## How to test this PR

- Run v2/automatedClaim with a claimId and Diagnostic code 7101
- Query the claim via v1/claimInfo. You should see the claim with one contention and summary
- Run v2/automatedClaim with the same claimId and different diagnostic code.
- Query the claim via v1/claimInfo. You should see the claim with both contentions.
 
[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
